runtime/compiler/control/HookedByTheJit.cpp

```diff
@@ -3146,8 +3146,13 @@ void jitIllegalFinalFieldModification(J9VMThread *currentThread, J9Class *fieldC
    reportHook(currentThread, "jitIllegalFinalFieldModification", "class %p %.*s", fieldClass, length, className);
 
    TR::CompilationInfo * compInfo = TR::CompilationInfo::get(jitConfig);
+
    TR_RuntimeAssumptionTable * rat = compInfo->getPersistentInfo()->getRuntimeAssumptionTable();
-   if (rat)
+   if (rat
+#if defined(J9VM_OPT_JITSERVER)
```

Comment:
If we take this path at the server something is wrong.
Thus, i would rather place an ASSERT_FATAL here.

Reply:
We do take this path at the server. Investigating what's wrong here.